src/renderer.cpp

```diff
@@ -1181,18 +1189,18 @@ void Renderer::setScissor(uint16_t x, uint16_t y, uint16_t width, uint16_t heigh
         }
 
         // This removes the black bars on the top and bottom of the screen
-        if(y = 16 && height == 448)
+        if(y == 16 && height == 448)
         {
             scissorOffsetY = getInternalCoordY(0.0);
             scissorHeight = getInternalCoordY(480);
         }
 
         // This sets a scissor offset for field with width not enough to fill the screen in 16:9
-        field_trigger_header* field_triggers_header_ptr = *ff7_externals.field_triggers_header;

```

Comment:
Better to rewrite this section. I would do it like this:
```cpp
struct game_mode* mode = getmode_cached();
switch(mode) {
   case MODE_CREDITS:
       // modify scissorOffsetX only
    case MODE_FIELD:
       // is_movie_playing
      
       // remove black bars

      // widescreen.getMode(), In your code I saw == WM_DISABLED, is it correct?
   default:
      // maybe do not apply scissorOffsetX shift only if x == 0 and game_width == 640 or find another way.
}

```

Reply:
I agree, it makes the code much more cleaner.

Reply:
Rewrited this section with a switch statement and changed some conditions to fix known bugs.
As it is now framebuffer effects work fine and the limit break boxes are now fixed too!